### DISCUSSION TITLE
Are modules used in data fetching methods also bundled for client?

### DISCUSSION THREAD

**Discussion by michael-mcgreal** [vercel/next.js#40274] [Help] [answered]
I am importing a large prisma client into a pages file, that is used in the getStaticPaths data fetching method only. 

Will this import be shipped to the client also? 

_(or does nextjs do something like tree-shaking server-only imports when building?)_ 

(I think the answer is that the modules are not bundled for the client, but I am unsure how to test). Thank you.

___ 
related discussions: https://github.com/vercel/next.js/discussions/17053#discussion-19122

**Comment by gaston-flores** [comment, marked as answer]
No, modules used only for server-side fetching [are not included in the client bundle](https://nextjs.org/docs/basic-features/data-fetching/get-static-props#write-server-side-code-directly). You can test this out [using this tool.](https://next-code-elimination.vercel.app/)

**Reply by michael-mcgreal** [reply]
Wow @gaston-flores superb feature and tool. Much appreciated.

**Reply by bennettdams** [reply]
Does anyone know why `const`s defined in a file, but not used, are bundled for the client as well? Is there a way to prevent this?
 See: https://next-code-elimination.vercel.app/s/tI1txBWYk

![image](https://user-images.githubusercontent.com/29319414/204586352-3262eac1-b4a6-4406-a082-0f30b3a9d5a8.png)
